File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxLevelSum = function(root) {
    
    if (root === null) {
      return -1;
    }

    let maxLevel = 1;
    let maxSum = root.val; 
    let level = 1;

    const queue = [];
    queue.push(root);

    while (queue.length > 0) {
      const levelSize = queue.length;
      let levelSum = 0;

      for (let i = 0; i < levelSize; i++) {
        const curr = queue.shift();

        levelSum += curr.val;

        if (curr.left !== null) {
          queue.push(curr.left);
        }

        if (curr.right !== null) {
          queue.push(curr.right);
        }
      }

      if (levelSum > maxSum) {
        maxSum = levelSum;
        maxLevel = level;
      }

      level++;
    }

    return maxLevel;
  
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        if (root == nullptr) {
            return -1;
        }

        int maxLevel = 1;
        int maxSum = root->val; // Initialize with the root value
        int level = 1;

        std::queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            int levelSize = q.size();
            int levelSum = 0;

            for (int i = 0; i < levelSize; ++i) {
                TreeNode* curr = q.front();
                q.pop();

                levelSum += curr->val;

                if (curr->left != nullptr) {
                    q.push(curr->left);
                }

                if (curr->right != nullptr) {
                    q.push(curr->right);
                }
            }

            if (levelSum > maxSum) {
                maxSum = levelSum;
                maxLevel = level;
            }

            ++level;
        }

        return maxLevel;
    }
};